pkg/sdk/sdk_test.go

```diff
@@ -0,0 +1,503 @@
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: json failure",
+			id:   "1",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register/{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+
+					json.NewEncoder(w).Encode("")
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "json: cannot unmarshal string into Go value of type model.Response" {
+					t.Errorf("Want: %v, Got: %v", "json: cannot unmarshal string into Go value of type model.Response", err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: incorrect id in response",
+			id:   "1",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register/{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+					vars := mux.Vars(r)
+					_, ok := vars["id"]
+					if !ok {
+						response.ToJson(w, http.StatusBadRequest, "id not found", nil)
+						return
+					}
+					err := r.ParseMultipartForm(10000)
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "file size exceeeded", nil)
+						log.Error(err.Error())
+						return
+					}
+					file, _, err := r.FormFile("file")
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "file parse fail", nil)
+						log.Error(err.Error())
+						return
+					}
+					defer file.Close()
+
+					response.ToJson(w, http.StatusOK, "SUCCESS", map[string]interface{}{
+						"id": "gfhgv",
+					})
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "incorrect id received in response" {
+					t.Errorf("Want: %v, Got: %v", "incorrect id received in response", err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: incorrect status code",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+					err := r.ParseMultipartForm(10000) //File size to come from config
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "file size exceeded", nil)
+						log.Error(err.Error())
+						return
+					}
+					file, _, err := r.FormFile("file")
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "file parse fail", nil)
+						log.Error(err.Error())
+						return
+					}
+					defer file.Close()
+
+					response.ToJson(w, http.StatusCreated, "SUCCESS", map[string]interface{}{
+						"id": gomock.Any(),
+					})
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "non success status code received : 404" {
+					t.Errorf("Want: %v, Got: %v", "non success status code received : 404", err.Error())
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: incorrect file type", //doubt
```

Comment:
Not sure how this is a relevant test case seeing as the files can be of any type with no way of determining if they are correct before the pdf is being generated. Remove if not relevant/required.

Reply:
done